tests/image/test_deploy.py

```diff
@@ -130,17 +118,25 @@ def _load_deploy_event_result(line):
     return json.loads(line.replace("'", '"'))
 
 
-@pytest.mark.usefixtures('project_dir', 'mocked_post')
-@mock.patch('requests.get')
-@mock.patch('shub.image.list.list_cmd')
-def test_progress_verbose_logic(list_mocked, mocked_get, monkeypatch):
-    monkeypatch.setattr('shub.image.deploy.SYNC_DEPLOY_REFRESH_TIMEOUT', 0)
-    list_mocked.return_value = ['a1f', 'abc', 'spi-der']
-    mocked_get.side_effect = DEPLOY_RESPONSES_BASE_SAMPLE
+def _get_logged_progress_cmd(progress_cls, events):
+    @command()
+    def progress_cmd():
+        progress = progress_cls(events)
+        progress.show()
+    return progress_cmd
 
-    result = CliRunner().invoke(cli, ["dev", "--version", "test", "--verbose"])
-    assert result.exit_code == 0
 
+def _yield_events_with_timeout(events, timeout):
+    for event in events:
+        yield event
+        time.sleep(timeout)
```

Comment:
This timeout is directly correlated with `tqdm` output, if you remove it - it will consume events instantly and print only initial (0%) and final (100%) progress instead of all the steps, I believe it's worth to keep the timeout to be closer to real-life conditions.